src/coordinator/functions/count.go

```diff
@@ -57,22 +58,28 @@ type CountNode struct {
 }
 
 // Process the block
-func (c *CountNode) Process(ID parser.NodeID, block storage.Block) error {
+func (c *CountNode) Process(ID parser.NodeID, block block.Block) error {
 	builder, err := c.controller.BlockBuilder(block.Meta())
 	if err != nil {
 		return err
 	}
 
 	stepIter := block.StepIter()
+	if err := builder.AddCols(stepIter.Steps()); err != nil {
+		return err
+	}
+
 	for index := 0; stepIter.Next(); index++ {
 		step := stepIter.Current()
 		values := step.Values()
-		sum := 0.0
+		count := 0.0
 		for _, value := range values {
-			sum += value
+			if !math.IsNaN(value) {
+				count++
+			}
 		}
 
-		builder.AppendValue(index, sum)
+		builder.AppendValue(index, count)
 	}
 
 	return c.controller.Process(builder.Build())
```

Comment:
This function does not seem to conform with our design where we would reuse the input block as the "output". Would make more sense to have the block columns have a Reset() method or something, and set that to `[]int{count}`

Reply:
so there are cases when we can't use input block. Eg: if input block is compressed, it won't have any slices allocated. I would suggest that we put the logic of pooling blocks within builder and individual functions don't have any idea about where the blocks are coming from.